src/aiodynamo/fast/client.py

```diff
@@ -0,0 +1,613 @@
+            key_condition, True
+        )
+        expression_attribute_names.update(ean)
+        expression_attribute_values.update(eav)
+
+        payload = {
+            "TableName": table,
+            "KeyConditionExpression": key_condition_expression,
+            "ExpressionAttributeNames": expression_attribute_names,
+            "ExpressionAttributeValues": py2dy(expression_attribute_values),
+            "Select": Select.count.value,
+        }
+        if start_key:
+            payload["ExclusiveStartKey"] = start_key
+        if filter_expression:
+            payload["FilterExpression"] = filter_expression
+        if index:
+            payload["IndexName"] = index
+        count_sum = 0
+        async for result in self._fast_depaginate("Query", payload):
+            count_sum += result["Count"]
+        return count_sum
+
+    async def update_item(
+        self,
+        table: TableName,
+        key: Item,
+        update_expression: UpdateExpression,
+        *,
+        return_values: ReturnValues = ReturnValues.none,
+        condition: ConditionBase = None,
+    ) -> Union[Item, None]:
+
+        (
+            update_expression,
+            expression_attribute_names,
+            expression_attribute_values,
+        ) = update_expression.encode()
+
+        if not update_expression:
+            raise EmptyItem()
+
+        builder = ConditionExpressionBuilder()
+        if condition:
+            condition_expression, ean, eav = builder.build_expression(condition)
+            expression_attribute_names.update(ean)
+            expression_attribute_values.update(eav)
+        else:
+            condition_expression = None
+
+        payload = {
+            "TableName": table,
+            "Key": py2dy(key),
+            "UpdateExpression": update_expression,
+            "ExpressionAttributeNames": expression_attribute_names,
+            "ExpressionAttributeValues": py2dy(expression_attribute_values),
+            "ReturnValues": return_values.value,
+        }
+        if condition:
+            payload["ConditionExpression"] = condition_expression
+
+        resp = await self.send_request(action="UpdateItem", payload=payload)
+
+        if "Attributes" in resp:
+            return dy2py(resp["Attributes"], self.numeric_type)
+        else:
+            return None
+
+    async def send_request(
+        self, *, action: str, payload: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        key = await self.credentials.get_key(self.http)
+        request = signed_dynamo_request(
+            key=key,
+            payload=payload,
+            action=action,
+            region=self.region,
+            endpoint=self.endpoint,
+        )
+        return await self.http.post(
+            url=request.url, headers=request.headers, body=request.body
+        )
+
+    async def _fast_depaginate(self, action, payload) -> AsyncIterator[Dict[str, Any]]:
```

Comment:
it's a bit ugly that payload is changed in place, but I maybe it's ok for an internal function...
as long as there no reties and every caller creates a new payload each time, it's safe, right?

Reply:
that was the idea. it's at the very end and prefixed by `_` with a reason. Doing a `deepcopy` feels bad in a "fast" API.

Reply:
there's no need for **deep** copy though :)
anyway, I reckon it's acceptable, let's leave a gotcha for some poor future programmer :P